### PR TITLE
Fix real literal parsing in the odata query

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs [hunk 1]
```diff
@@ -8,6 +10,58 @@ namespace Microsoft.WindowsAzure.MobileServices.Test.UnitTests.OData
...
     public class ODataExpressionParserTest : TestBase
     {
+        class CurrentCultureHelper : IDisposable
+        {
+            private CultureInfo previous;
+
+            public static CultureInfo CurrentCulture
+            {
+#if NET45
+                get => Thread.CurrentThread.CurrentCulture;
```
Comment (dhei) [inline unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: `System.Globalization.CultureInfo.CurrentCulture` api exists in .NET 4.5.1 (https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentculture?view=netframework-4.5.1#System_Globalization_CultureInfo_CurrentCulture). Why using `Thread.CurrentThread.CurrentCulture`?

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs [hunk 2]
```diff
@@ -8,6 +10,58 @@ namespace Microsoft.WindowsAzure.MobileServices.Test.UnitTests.OData
...
+            {
+                CurrentCulture = previous;
+            }
+        }
+
+        [TestMethod]
+        public void ParseFilter_Real_NumberDecimalSeparator()
+        {
+            // Set some CultureInfo with different decimal separator
+            using (var _ = new CurrentCultureHelper("ru-RU"))
```
Comment (dhei) [inline unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: The customer reported issue is related to users in Finland, so can you also test `fi-FI`?

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs (hunk 1 above)
Comment (MatkovIvan) [reply unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: @dhei From this doc: `In the .NET Framework 4.5.2 and earlier versions, the CurrentCulture property is read-only;`

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs (hunk 2 above)
Comment (MatkovIvan) [reply unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: The idea is to have the different decimal separator. Any lang with different separator can be applied. I can change it here, but it doesn't make any sense to me.

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs (hunk 1 above)
Comment (dhei) [reply unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: Thanks Ivan

File: unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs (hunk 2 above)
Comment (dhei) [reply unittest/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/OData/ODataExpressionParser.Test.cs]: Thanks @MatkovIvan for the clarification. "ru-RU" or "fi-FI" use comma (`,`) as decimal separator while the default "en-US" uses dot (`.`) as decimal separator. So a test case on "ru-RU" is as good as one on "fi-FI".

**Pull request merged**